core/controllers/acl_decorators.py

```
@@ -5133,6 +5134,98 @@ def test_can_access_certificate_dashboard(
+    def test_can_access(
+        self: _SelfBaseHandlerType, question_id: str, **kwargs: Any
+    ) -> _GenericHandlerFunctionReturnType:
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+        attempt = certificate_assessment_services._get_active_attempt_for_learner(  # pylint: disable=protected-access
+            self.user_id
+        )
+        if question_id not in attempt.version_data['question_versions']:
+            raise self.InvalidInputException(
+                'Question is not part of this assessment.'
+            )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)